apps/src/applab/PlaySpaceHeader.jsx

```diff
@@ -35,12 +35,22 @@ var PlaySpaceHeader = React.createClass({
 
   render: function () {
     var leftSide, rightSide;
+    var showDataModeButton = Applab.useFirebase;
+    var toggleGroupWidth = showDataModeButton ? '160px' : '120px';
 
     if (!this.shouldHideToggle()) {
+      var toggleButtons = [
+        <button id='codeModeButton' value={ApplabInterfaceMode.CODE}>{msg.codeMode()}</button>,
+        <button id='designModeButton' value={ApplabInterfaceMode.DESIGN}>{msg.designMode()}</button>
+      ];
+      if (showDataModeButton) {
+        toggleButtons.push(
+          <button id='dataModeButton' value={ApplabInterfaceMode.DATA}>{msg.dataMode()}</button>
+        );
+      }
       leftSide = (
         <ToggleGroup selected={this.props.interfaceMode} onChange={this.props.onInterfaceModeChange}>
-          <button id='codeModeButton' value={ApplabInterfaceMode.CODE}>{msg.codeMode()}</button>
-          <button id='designModeButton' value={ApplabInterfaceMode.DESIGN}>{msg.designMode()}</button>
+          {toggleButtons}
```

Comment:
I tried the way you're suggesting and it didn't work, but now I realize that it was our own processing of these elements in `ToggleGroup.renderChildren` that choked on the undefined child, not React itself. Should be an easy fix.